File: reservation_express/lib/models/Order.dart
class Order {
  final int id;
  final int reservationId;
  final int userId;
  final int tableId;
  final String orderNumber;
  final DateTime orderDate;
  final double totalAmount;
  final String status;
  final String? itemsJson;
  final String? itemsSummary;

  Order({
    required this.id,
    required this.reservationId,
    required this.userId,
    required this.tableId,
    required this.orderNumber,
    required this.orderDate,
    required this.totalAmount,
    required this.status,
    this.itemsJson,
    this.itemsSummary,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id']?.toInt() ?? 0,
      reservationId: json['reservation'] != null ? (json['reservation']['id']?.toInt() ?? 0) : 0,
      userId: json['user'] != null ? (json['user']['id']?.toInt() ?? 0) : 0,
      tableId: json['table'] != null ? (json['table']['id']?.toInt() ?? 0) : 0,
      orderNumber: json['orderNumber'] ?? '',
      orderDate: json['orderDate'] != null 
          ? DateTime.parse(json['orderDate'])
          : DateTime.now(),
      totalAmount: (json['totalAmount'] as num?)?.toDouble() ?? 0.0,
      status: json['status'] ?? 'pending',
      itemsJson: json['itemsJson'],
      itemsSummary: json['itemsSummary'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'reservation': {'id': reservationId},
      'user': {'id': userId},
      'table': {'id': tableId},
      'orderNumber': orderNumber,
      'orderDate': orderDate.toIso8601String(),
      'totalAmount': totalAmount,
      'status': status,
      'itemsJson': itemsJson,
      'itemsSummary': itemsSummary,
    };
  }
}

File: Documents/workspace-spring-tools-for-eclipse-4.32.0.RELEASE/Reservation-express/src/main/java/model/Order.java
package model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "reservation_id", nullable = false)
    private Reservation reservation;
    
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @ManyToOne
    @JoinColumn(name = "table_id", nullable = false)
    private RestaurantTable table;
    
    @Column(name = "order_number", unique = true, nullable = false)
    private String orderNumber;
    
    @Column(name = "order_date")
    private LocalDateTime orderDate;
    
    @Column(name = "total_amount", nullable = false)
    private Double totalAmount;
    
    @Column(nullable = false)
    private String status = "pending"; // pending, confirmed, preparing, ready, served, cancelled
    
    // Constructors
    public Order() {
        this.orderDate = LocalDateTime.now();
        this.orderNumber = generateOrderNumber();
    }
    
    public Order(Reservation reservation, User user, RestaurantTable table, Double totalAmount) {
        this();
        this.reservation = reservation;
        this.user = user;
        this.table = table;
        this.totalAmount = totalAmount;
    }
    
    private String generateOrderNumber() {
        return "ORD-" + System.currentTimeMillis();
    }
    
    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    
    public Reservation getReservation() { return reservation; }
    public void setReservation(Reservation reservation) { this.reservation = reservation; }
    
    public User getUser() { return user; }
    public void setUser(User user) { this.user = user; }
    
    public RestaurantTable getTable() { return table; }
    public void setTable(RestaurantTable table) { this.table = table; }
    
    public String getOrderNumber() { return orderNumber; }
    public void setOrderNumber(String orderNumber) { this.orderNumber = orderNumber; }
    
    public LocalDateTime getOrderDate() { return orderDate; }
    public void setOrderDate(LocalDateTime orderDate) { this.orderDate = orderDate; }
    
    public Double getTotalAmount() { return totalAmount; }
    public void setTotalAmount(Double totalAmount) { this.totalAmount = totalAmount; }
    
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
}